### PR TITLE
ESLint & Bun extension support

### LINKED ISSUE #30: cvar test has duplicate keys
eslint doesn't like testcvar.js because some keys are duplicated
This wouldn't be a problem - except those definitions contradict eachother

for the output locally, create elint.config.js and populate it with (yoinked from [docs](https://eslint.org/docs/latest/use/getting-started#configuration))
```js
import js from "@eslint/js"; // bun i eslint

export default [
	js.configs.recommended,

	{
		rules: {
			"no-unused-vars": "off",
			"no-undef": "off"
		}
	}
];
```
and then `bun i eslint` (will add to package.json - i'd just discard unless you want ci/cd stuffs) now `bunx eslint` will complain about `util/testcvar.js`. Specifically

just ctrl+F in the file to find the line numbers
```js
// cvarTestSafe
  "sar_force_fov": (val) => true, // 	functional 	-
  "sar_force_fov": (val) => val >= 45 && val <= 140, // 	functional 	45 .. 140

  "sar_stop": (val) => true, // 	non-functional 	-
  "sar_stop": (val) => false, // 	functional 	-

  "sar_stats_auto_reset": (val) => true,

  "sar_toast_create": (val) => true,
  "sar_toast_create": (val) => true, // 	non-functional 	-

// cvarTestIllegal
  "sar_aircontrol": (val) => true,
  
  "sar_autojump": (val) => true,
  "sar_autojump": (val, cheats) => cheats,

  "sar_duckjump": (val) => true,

  "ent_fire": (val) => true,
  
  "notarget": (val, cheats) => cheats,
  "notarget": (val) => true,
```

### LINKED ISSUE #67: Bundle ESLint and Bun configurations with epochtal
One at a time:

# Bun launch configurations
Installing the official VSCode Bun extension makes debugger a whole lot easier.

Immediately, breakpoints and proper debugging becomes possible. The purpose of debugging should be self explanatory, however here's two screenshots to show just how neat the debugger is:
image
image

Additionally, one can FINALLY bind restarting epochtal to a hotkey, *woo*. All of this can be achieved by bundling a single launch.json file that completely universal.

# ESLint
VSCode's integrated JavaScript language server simply sucks. Bugs like steamid and the countless other ones I encountered after attaching ESLint should've been avoided from the start.

ESLint is the go-to js language server. It's configured through a single json file and completely customizable. It can immediately highlight unused variables, potential undeclared variables and most importantly: tell you where you forgot semicolons, lol. It can also enforce double quotes, curly braces or whatever you program it to. After some tampering around I achieved a config that already discovered 3 enormous bugs, one of which I will post below:
image
image

Setting up ESLint again is just a VSCode extension. On Arch Linux I didn't even have to install the system package `eslint`, though that might be necessary on other distros, idk.

I've already started working on a configuration, so I will immediately assign myself to this.
(also, I did not find any of the tags suitable for this)

### PR DESCRIPTION
**Pull request by PancakeTAS**
This PR adds an epochtal-tailored ESLint configuration and fixes issues addressed by it.
This PR also adds Bun files for quick launching of epochtal.

I will make a very short video outlining how to set stuff up (once the PR is ready), because VSCode makes this incredibly confusing.

Closes #67
Closes #30 
Let's see if this works: Closes #55

### CODE REVIEW

**Comment by PancakeTAS** [comment]
Okay I think it's ready. I added basic syntax checking for the frontend, because adding actual error highlighting and stuff just straight up does not work with the frontend because of it's incredibly weird global scope.
The spplice file has proper error checking again though, lol.

Also screw making a video, it'll go 10x quick if I just write it in text.
0. Install `eslint` and `eslint-language-server` (these are Arch packages, not necessary but they help get a summary of the whole codebase in one console command)
1. Install extensions "ESLint" by Microsoft and "Bun" by Oven (not by Pandy!).
2. Switch to this branch.. or main, if it's merged
3. Run `bun i` to install the deps
4. Open commands window (ctrl shift p) and run "ESLint: Restart ESLint Server"
5. Mess up the code and check highlighted code or the "Problems" panel

If you wanna run Epochtal:
1. Head to the "Run and Debug" sidebar
2. Select "Run Epochtal" in the dropdown at the top
3. Click the run icon right next to it

Once Epochtal is running, you can do these things:
- In the "Run and Debug" sidebar:
  - "Watch" section: Watch variables
  - "Variables" section: When breakpointed or paused, see all variables in the local scope
  - "Call Stack" section: Shows every thread and it's current function when paused
  - "Breakpoints" section: Toggle when to breakpoint and toggle self-set breakpoints
- In the Editor:
  - Set breakpoints by clicking left of the line number.
  - When paused, hover over variables to see their value (doesn't work on every variable)
  - When paused, use the top bar to step/pause/unpause/restart epochtal
- In the "Debug Console" panel (it's next to problems and terminal):
  - Type the name of a global to fully print it, like `epochtal.data`
  - Read sysout of the process.

If you wanna bind starting and restarting Epochtal to a key, you have to bind `Debug: Restart` aka `workbench.action.debug.restart` in the settings and not any of the other ones.

File: util/profiledata.js [hunk 1]
```diff
@@ -85,7 +86,7 @@ module.exports = async function (args, context = epochtal) {
 
       if (profilesPath) {
         const dataPath = `${profilesPath}/${steamid}/data.json`;
-        await Bun.write(dataPath, JSON.stringify(profile));
+        if (fs.existsSync(dataPath)) await Bun.delete(dataPath);
```
Comment (p2r3) [inline util/profiledata.js:89]: are we... sure this is how you delete a file with bun? i thought there was no bun api for that:
https://bun.sh/guides/write-file/unlink

File: util/demo.js [hunk 2]
```diff
@@ -212,15 +211,15 @@ module.exports = async function (args, context = epochtal) {
 
       let ppnf = false, sv_cheats = false;
       let lastTimestamp = null, speedrunTimer = null;
-      let timescale = [], timescaleTotal = 0;
+      const serverTimestamp = Date.now();
```
Comment (p2r3) [inline util/demo.js:214]: this _reeeally_ should be something like `referenceTimestamp`, or pretty much anything that would resolve the ambiguity of referring to two different things with the name "server timestamp"

File: util/workshopper.js [hunk 3]
```diff
@@ -76,7 +76,7 @@ async function curateWorkshop (maps = []) {
   while (startDate - lastDate < weekSeconds) {
 
     // Fetch page of workshop data
-    const request = await fetch(`${requestData}&page=${page}`);
+    await fetch(`${requestData}&page=${page}`); // > Is this necessary? - PancakeTAS
```
Comment (p2r3) [inline util/workshopper.js:79]: no

**Review by p2r3 — CHANGES_REQUESTED** [review]

**Review by p2r3 — APPROVED** [review]

**Pull request merged**